doc/src/guides/solving/solving-guidance.md

```diff
@@ -0,0 +1,221 @@
+# Solving Guidance
+
+These guidelines apply to many types of solving.
+
+## Numeric Solutions
+
+### Numeric Solutions by Choice
+Solving functions such as {func}`~.solve` and {func}`~.solveset` will not try to
+find a numeric solution, only a mathematically-exact symbolic solution. So if
+you want a numeric solution, consider {func}`~.nsolve`.
+
+In some situations, even though a closed-form solution is available, it may be
+too cumbersome to be desirable. In that case, you can use {func}`evalf n
+<sympy.core.evalf>` instead if a numerical solution is acceptable:
+
+```py
+>>> from sympy import symbols, solve
+>>> x = symbols('x')
+>>> solutions = solve(x**4 + 10*x**2 + x + 1, x, dict=True); solutions
+[{x: -sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3))/2 - sqrt(-40/3 - 2*(1307/432 + sqrt(434607)*I/144)**(1/3) + 2/sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3)) - 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)))/2}, {x: sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3))/2 - sqrt(-40/3 - 2*(1307/432 + sqrt(434607)*I/144)**(1/3) - 2/sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3)) - 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)))/2}, {x: sqrt(-40/3 - 2*(1307/432 + sqrt(434607)*I/144)**(1/3) - 2/sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3)) - 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)))/2 + sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3))/2}, {x: sqrt(-40/3 - 2*(1307/432 + sqrt(434607)*I/144)**(1/3) + 2/sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3)) - 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)))/2 - sqrt(-20/3 + 56/(9*(1307/432 + sqrt(434607)*I/144)**(1/3)) + 2*(1307/432 + sqrt(434607)*I/144)**(1/3))/2}]
+>>> for solution in solutions:
+...     print(solution[x].n())
+-0.0509758447494279 + 0.313552108895239*I
+0.0509758447494279 + 3.14751999969868*I
+0.0509758447494279 - 3.14751999969868*I
+-0.0509758447494279 - 0.313552108895239*I
+```
+
+### Equations With no Analytical Solution
+
+The vast majority of arbitrary nonlinear equations are not analytically
+solvable. The classes of equations that are solvable are basically:
+1. Linear equations
+2. Polynomials, except where limited by the Abel-Ruffini theorem
+3. Equations that can be solved by inverting some transcendental functions
+4. Problems that can be transformed into the cases above (e.g., by turning
+trigonometric functions into polynomials)
+5. A few other special cases that can be solved with something like the
+{class}`Lambert W function <sympy.functions.elementary.exponential.LambertW>` 
+
+SymPy may reflect that your equation has no solutions that can be expressed
```

Comment:
Just to be clear, this could mean it can't be solved in closed-form or it could just mean that it isn't implemented, as the error name suggests. SymPy doesn't have any functionality to guarantee that an equation's solutions cannot be written in closed-form (in general, that's not a very easy thing to prove).